src/components/ArticlePage/index.jsx

```diff
@@ -9,32 +9,50 @@ import RatingStarsBox from '../../components/RatingStarsBox';
 import HorizontalLine from '../../components/HorizontalLine';
 import TextAreaInput from '../../components/TextAreaInput';
 import CommentEntries from '../../components/CommentEntries';
-import { getOneArticle } from '../../redux/actions/articles';
+import { getOneArticle, rateArticle, getArticleAvgRating } from '../../redux/actions/articles';
+import { toast } from 'react-toastify';
 import './style.scss';
 
 
 export class SingleArticleView extends Component {
 
-  componentDidMount() {
-    const { fetchArticle, match } = this.props;
+  async componentDidMount() {
+    const { fetchArticle, getArticleAvgRating, match } = this.props;
     let articleId = match.params.articleId;
-    fetchArticle(articleId);
+    await fetchArticle(articleId);
+    getArticleAvgRating(articleId)
+  }
+
+  starClickHandle(item) {
+    let articleId = this.props.match.params.articleId;
+    const _this = this;
+    this.props.rateArticle(item, articleId )
+    .then(() => toast.success('Thank you for rating this article'))
+    .catch(function errorHandler(error) {
+      if (error.response.data.message === 'You already rated this article'){
+        return toast.error('You have already rated this article');
+      } else if(error.response.data === 'Unauthorized'){
+         toast.error('You need to login to rate an article');
+        return  _this.props.history.push('/login')
+      }
+      toast.error('Error while rating article make sure you are logged in');
+    })
   }
 
   render() {
     const { match, articles = [] } = this.props;
     const { articleId } = match.params;
     const article = articles.find(item => item.id === articleId) || {};
-
     const { 
       title, 
       featuredImage = '/assets/img/map-typing.jpg', 
       author: { username = '', imageUrl: userAvatarImage = '' } = {}, 
```

Comment:
@Princesso Please explain what `line 49` is meant to do.

Reply:
I did not work on this section of the single article view page. @jsamchineme  may be able to provide context to what this does.